pkg/arvo/sys/hoon.hoon

```diff
@@ -11592,37 +11612,61 @@
         ==
       ==
     ::
+    ::  +dibs: parses a $link at the start of a formal comment
+    ++  dibs
+      %+  cook  |=([a=term b=term] ;;(link [a b]))
+      %-  stew
+      ^.  stet  ^.  limo
+      :~  :-  '|'
+          ;~(pfix bar (stag %chat sym))
+          :-  '.'
+          ;~(pfix dot (stag %frag sym))
+          :-  '+'
+          ;~(pfix lus (stag %funk sym))
+          :-  '$'
+          ;~(pfix buc (stag %grog sym))
+          :-  ' '
+          ;~(pfix step (stag %funk (easy %$)))
+      ==
+    ++  shot
+      ;~(plug ;~(sfix line (just `@`10) (punt gap)) (rant ;~(less tine text)))
+    ::
```

Comment:
I knew the parser for batch comments wasn't quite right - looking at it again today, I'm pretty sure I made use of the `def` face in `$whit` incorrectly. I was thinking that that would be where things like docs about the sample of the gate would go. That's true, but I think it's intended for batch comments in general. Right now I have split doccords parsing into two arms, `apex:docs` and `vext:docs` with the latter parsing a `(list whit)` intended for batch comments and the former just a `whit`. But looking to the future of how batch comment parsing ought to work, there should be only one function parsing a single `whit`. I think the new definition of `whit` would look like

```
+$  whit
  $:  lab=(list link)
       boy=(unit (pair cord (list sect))
       def=(map (list link) whit)
       use=(set (list link))
  ==
```
Or maybe `def=(map (list link) (pair cord (list sect)))`. Either way I knew I needed to switch to lists of links, but I need to rewrite more of the batch comment stuff than I had expected for this to be suitable for release.

Reply:
https://github.com/urbit/urbit/pull/5873/commits/de19456f82b03b41d952b36b5c73b07e1f21ed18